filters/src/main/java/io/camunda/zeebe/process/test/filters/logger/RecordStreamLogger.java

```diff
@@ -0,0 +1,291 @@
+import io.camunda.zeebe.protocol.record.value.MessageStartEventSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.MessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessEventRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceCreationRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceResultRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessMessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.TimerRecordValue;
+import io.camunda.zeebe.protocol.record.value.VariableDocumentRecordValue;
+import io.camunda.zeebe.protocol.record.value.VariableRecordValue;
+import io.camunda.zeebe.protocol.record.value.deployment.ProcessMetadataValue;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RecordStreamLogger {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RecordStreamLogger.class);
+
+  private final RecordStream recordStream;
+  private final Map<ValueType, Function<Record<?>, String>> valueTypeLoggers = new HashMap<>();
+
+  public RecordStreamLogger(final RecordStreamSource recordStreamSource) {
+    this.recordStream = RecordStream.of(recordStreamSource);
+    valueTypeLoggers.put(ValueType.JOB, this::logJobRecord);
+    valueTypeLoggers.put(ValueType.DEPLOYMENT, this::logDeploymentRecord);
+    valueTypeLoggers.put(ValueType.PROCESS_INSTANCE, this::logProcessInstanceRecord);
+    valueTypeLoggers.put(ValueType.INCIDENT, this::logIncidentRecord);
+    valueTypeLoggers.put(ValueType.MESSAGE, this::logMessageRecord);
+    valueTypeLoggers.put(ValueType.MESSAGE_SUBSCRIPTION, this::logMessageSubscriptionRecord);
+    valueTypeLoggers.put(
+        ValueType.PROCESS_MESSAGE_SUBSCRIPTION, this::logProcessMessageSubscriptionRecord);
+    valueTypeLoggers.put(ValueType.JOB_BATCH, this::logJobBatchRecord);
+    valueTypeLoggers.put(ValueType.TIMER, this::logTimerRecord);
+    valueTypeLoggers.put(
+        ValueType.MESSAGE_START_EVENT_SUBSCRIPTION, this::logMessageStartEventSubscriptionRecord);
+    valueTypeLoggers.put(ValueType.VARIABLE, this::logVariableRecord);
+    valueTypeLoggers.put(ValueType.VARIABLE_DOCUMENT, this::logVariableDocumentRecord);
+    valueTypeLoggers.put(
+        ValueType.PROCESS_INSTANCE_CREATION, this::logProcessInstanceCreationRecord);
+    valueTypeLoggers.put(ValueType.ERROR, this::logErrorRecord);
+    valueTypeLoggers.put(ValueType.PROCESS_INSTANCE_RESULT, this::logProcessInstanceResultRecord);
+    valueTypeLoggers.put(ValueType.PROCESS, this::logProcessRecord);
+    valueTypeLoggers.put(ValueType.PROCESS_EVENT, this::logProcessEventRecord);
+  }
```

Comment:
I understand why you'd say this. I didn't include all value types because these value types don't provide any extra information that would be useful for a user.

I've added the test but I do think it's gonna be an annoying situation for release managers. After all that is the moment where we will update the Zeebe version and encounter the failing tests.

Reply:
>it's gonna be an annoying situation for release managers

Yes, that is the cost of this approach. The alternative is that users might see the generic record logger's output + us not being aware of it.

With the test in place, the worst case is that we have to temporarily ignore the test to release a version of zeebe-process-test. Right?

Reply:
Yea, either that or just adding the value type logger which isn't that much work either